### PR TITLE
Revise 0280 - Adding new parameter of requires audio support and bluetoothDeviceAddress

### PR DESCRIPTION
**Pull request by Akihiro-Miyazaki**
## Introduction

This is an update of SDL-0280.
Add "JavaScript Suite" in Impacted platforms.

## Motivation

Add the description of other platforms(such as iOS and JavaScript Suite).

## Proposed solution

Add the comment "This flow is only for Android devices.".
Replace the figure of registeringApp_flow.
 1. Add the sequence of RPC OnDeviceStateChanged.
 2. Add the path to the conventional sequence.
 3. Add the `MediaStreamingStatus.isAudioOutputAvailable`.
Add `BluetoothInfo` to `OnDeviceStateChanged`.

## Impact on existing code

Add "JavaScript Suite" to impacted platforms.

### CODE REVIEW

**Comment by Akihiro-Miyazaki** [comment]
Hi, @theresalech -san.
This documentation was revised according to [this comment](https://github.com/smartdevicelink/sdl_evolution/issues/941#issuecomment-675669406).
Could you please review my proposal?
Thank you.

File: proposals/0280-Adding-new-parameter-of-requiresAudioSupport-and-BluetoothDeviceAddress.md [hunk 1]
```diff
@@ -20,12 +20,15 @@ To solve this problem, add `bluetoothDeviceAddress` and `requiresAudioSupport` s
...
+Note: In the case of other platform (such as iOS and JavaScript Suite), the system can output the audio via USB. Therefore, other platform can omit the requiresAudioSupport parameter.
+
 ### Change the app registration flow
 
 In order for the app to be always registered, the changes in SDL session establishment to app registration flow are shown below.
+This flow is only for Android devices.
 If the SDL app or HU cannot use this function, it will perform the same operation as before.
 
-![registeringApp_flow](../assets/proposals/0280-Adding-new-parameter-of-requiresAudioSupport-and-BluetoothDeviceAddress/registeringApp_flow1.png)
+![registeringApp_flow](../assets/proposals/0280-Adding-new-parameter-of-requiresAudioSupport-and-BluetoothDeviceAddress/registeringApp_flow.png)
```
Comment (theresalech) [inline proposals/0280-Adding-new-parameter-of-requiresAudioSupport-and-BluetoothDeviceAddress.md:31]: In the top right corner of the image, you've added the below. Can you please advise which requested revision (from [this comment](https://github.com/smartdevicelink/sdl_evolution/issues/941#issuecomment-675669406)) required this change? 

Screen Shot 2020-09-08 at 1 27 02 PM

**Review by theresalech — CHANGES_REQUESTED** [review]
Hi @Akihiro-Miyazaki, thanks for your PR. I've left one question, and some suggested changes. Please let me know if you have any questions or concerns with addressing my comments. Thank you!

File: proposals/0280-Adding-new-parameter-of-requiresAudioSupport-and-BluetoothDeviceAddress.md [hunk 2]
```diff
@@ -20,12 +20,15 @@ To solve this problem, add `bluetoothDeviceAddress` and `requiresAudioSupport` s
 The current SDL Java Suite library cancels the transport connection if the `requiresAudioSupport` setting is TRUE and BT A2DP is not connected.
 However, with this proposal, by adding a new parameter, the SDL app always establishes the transport connection and is registered without depending on the connection status of BT A2DP. If BT A2DP is not connected, the HU will automatically connect BT using `bluetoothDeviceAddress` or request connection from the user.
 
+Note: In the case of other platform (such as iOS and JavaScript Suite), the system can output the audio via USB. Therefore, other platform can omit the requiresAudioSupport parameter.
```
[the remark this answers is not in the document]
Comment (Akihiro-Miyazaki) [reply proposals/0280-Adding-new-parameter-of-requiresAudioSupport-and-BluetoothDeviceAddress.md]: @theresalech -san,

> Please move this note to the Proposed solution section as requested in the Steering Committee decision.

Thank you for your advice.
However, Line 23 is in the `Proposed solution` section.
Could you please teach me what is wrong?

File: proposals/0280-Adding-new-parameter-of-requiresAudioSupport-and-BluetoothDeviceAddress.md (hunk 1 above)
Comment (Akihiro-Miyazaki) [reply proposals/0280-Adding-new-parameter-of-requiresAudioSupport-and-BluetoothDeviceAddress.md:31]: @theresalech -san,
This change is referred below.
I think this change is referred "Update to use flow chart".
What you say is to add OnDeviceStateChanged to the HMI API. And it is not need to add the flow chart.
Is it correct?
If yes, I will modify to delete new part from the flow chart.

>  Update to use flow chart in this comment. Also, add BluetoothInfo struct to OnDeviceStateChanged, with bluetoothDeviceAddress and a2dpConnectionState params as described in this comment,

Comment (theresalech) [reply proposals/0280-Adding-new-parameter-of-requiresAudioSupport-and-BluetoothDeviceAddress.md:31]: @Akihiro-Miyazaki, I see. The chart is okay as is then. Thanks!

File: proposals/0280-Adding-new-parameter-of-requiresAudioSupport-and-BluetoothDeviceAddress.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (theresalech) [reply proposals/0280-Adding-new-parameter-of-requiresAudioSupport-and-BluetoothDeviceAddress.md]: My mistake. Yes, this is in the correct section.

File: proposals/0280-Adding-new-parameter-of-requiresAudioSupport-and-BluetoothDeviceAddress.md [hunk 3]
```diff
@@ -20,7 +20,7 @@ To solve this problem, add `bluetoothDeviceAddress` and `requiresAudioSupport` s
 The current SDL Java Suite library cancels the transport connection if the `requiresAudioSupport` setting is TRUE and BT A2DP is not connected.
 However, with this proposal, by adding a new parameter, the SDL app always establishes the transport connection and is registered without depending on the connection status of BT A2DP. If BT A2DP is not connected, the HU will automatically connect BT using `bluetoothDeviceAddress` or request connection from the user.
 
-Note: In the case of other platform (such as iOS and JavaScript Suite), the system can output the audio via USB. Therefore, other platform can omit the requiresAudioSupport parameter.
+Note: In the case of other platforms (such as iOS and JavaScript Suite), the system can output the audio via USB. Therefore, other iOS and JavaScript Suite can omit the `requiresAudioSupport` parameter.
```
Comment (theresalech) [inline proposals/0280-Adding-new-parameter-of-requiresAudioSupport-and-BluetoothDeviceAddress.md:23]:
```suggestion
Note: In the case of other platforms (such as iOS and JavaScript Suite), the system can output the audio via USB. Therefore, iOS and JavaScript Suite can omit the `requiresAudioSupport` parameter.
```

**Review by theresalech — CHANGES_REQUESTED** [review]
I've suggested one more small change. After this is committed, I'll approve this PR. Thank you!

**Comment by Akihiro-Miyazaki** [comment]
Hi, @theresalech  -san.
Thank you in advance for your review.
I made a commit all of your suggestion.
Please confirm them.
Thank you.

**Review by theresalech — APPROVED** [review]

**Pull request merged**